system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/RunAsJarSystemTest.java

```diff
@@ -0,0 +1,150 @@
+    protected static final org.apache.mesos.elasticsearch.systemtest.Configuration TEST_CONFIG = new org.apache.mesos.elasticsearch.systemtest.Configuration();
+    private static final int NUMBER_OF_TEST_TASKS = 1;
+
+    // Need full control over the cluster, so need to do all the lifecycle stuff.
+    private MesosCluster cluster;
+    private static DockerClient dockerClient = DockerClientFactory.build();
+    private JarScheduler scheduler;
+
+    @Before
+    public void before() {
+        ClusterArchitecture.Builder builder = new ClusterArchitecture.Builder()
+                .withZooKeeper()
+                .withMaster()
+                .withSlave() // Have to have a slave before we build. Bit of a minimesos bug. This offer wont get accepted.
+                .withContainer(zkContainer -> new JarScheduler(dockerClient, zkContainer), ClusterContainers.Filter.zooKeeper());
+        scheduler = (JarScheduler) builder.build().getClusterContainers().getOne(container -> container instanceof JarScheduler).get();
+        IntStream.range(0, NUMBER_OF_TEST_TASKS).forEach(dummy ->
+            builder.withSlave(zooKeeper -> new MesosSlaveWithSchedulerLink(dockerClient, zooKeeper, scheduler))
+        );
+        cluster = new MesosCluster(builder.build());
+
+        cluster.start();
+    }
+
+    @After
+    public void stopContainer() {
+        cluster.stop();
+    }
+
+    @Rule
+    public final TestWatcher WATCHER = new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+            cluster.stop();
+        }
+    };
+
+    @BeforeClass
+    public static void prepareCleanDockerEnvironment() {
+        new DockerUtil(dockerClient).killAllSchedulers();
+        new DockerUtil(dockerClient).killAllExecutors();
+    }
+
+    @AfterClass
+    public static void killAllContainers() {
+        new DockerUtil(dockerClient).killAllSchedulers();
+        new DockerUtil(dockerClient).killAllExecutors();
+    }
+
+    @Test
+    public void shouldStartScheduler() throws IOException, InterruptedException {
+        ESTasks esTasks = new ESTasks(TEST_CONFIG, scheduler.getIpAddress());
+        new TasksResponse(esTasks, NUMBER_OF_TEST_TASKS);
+
+        ElasticsearchNodesResponse nodesResponse = new ElasticsearchNodesResponse(esTasks, NUMBER_OF_TEST_TASKS);
+        assertTrue("Elasticsearch nodes did not discover each other within 5 minutes", nodesResponse.isDiscoverySuccessful());
+    }
+
+    private static class JarScheduler extends AbstractContainer {
+        private final ZooKeeper zooKeeperContainer;
+        private String docker0AdaptorIpAddress;
+
+        protected JarScheduler(DockerClient dockerClient, ZooKeeper zooKeeperContainer) {
+            super(dockerClient);
+            this.zooKeeperContainer = zooKeeperContainer;
+            docker0AdaptorIpAddress = dockerClient.versionCmd().exec().getVersion().startsWith("1.9.") ? "172.17.0.1" : "172.17.42.1";
```

Comment:
I guess the idiot couldn't foresee the future and didn't want to write code for it.